src/styled.js

```diff
@@ -27,18 +33,25 @@ const styled = ({tagName, elementStyle, mountSheet}: StyledArgs) => {
   return class StyledElement extends Component {
     static tagName: string = tagName
     static style: ComponentStyleType = elementStyle
+    static contextTypes = {
+      [channel]: PropTypes.object
+    }
```

Comment:
> how do you get theme then?

please check some tests here to see how it works: https://github.com/cssinjs/styled-jss/pull/35/files#diff-0eb25c8c0f58b242378f9559b0b673d4R137

so I'm getting theme here: https://github.com/cssinjs/styled-jss/pull/35/files#diff-833acfbc067801c8b867e72a3869d85fR54

and updating via state here: https://github.com/cssinjs/styled-jss/pull/35/files#diff-833acfbc067801c8b867e72a3869d85fR82

so I'm merging props and state here to provide `theme` for function values in styles: https://github.com/cssinjs/styled-jss/pull/35/files#diff-833acfbc067801c8b867e72a3869d85fR96